### PR TITLE
Introduce Integration Patterns DAO

### PR DESCRIPTION
**Pull request by iocanel**
This pull reuqest adds a readonly DAO for integration patterns. 

Patterns are loaded from the funktion model. (stepkinds -> integration patterns).

Along with this the pull request also fixes 3 side issues:
- broken injection of cachecontainer
- broken injection of list of daos
- fetchAll not working correctly.

### CODE REVIEW

File: api/src/main/java/com/redhat/ipaas/api/v1/rest/CacheManagerProducer.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+        GlobalConfiguration g = new GlobalConfigurationBuilder()
+            .clusteredDefault()
+            .transport()
+            .clusterName("InfinispanCluster")
+            .build();
+
+        Configuration cfg = new ConfigurationBuilder()
+            .eviction()
+            .strategy(EvictionStrategy.LRU)
+            .maxEntries(150)
```
Comment (jimmidyson) [inline api/src/main/java/com/redhat/ipaas/api/v1/rest/CacheManagerProducer.java]: Can we make this configurable?

File: api/src/main/java/com/redhat/ipaas/api/v1/rest/IntegrationPatternDAO.java [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+    @Override
+    public ListResult<IntegrationPattern> fetchAll() {
+        return new ListResult.Builder<IntegrationPattern>()
+            .items(PATTERNS.values())
+            .totalCount(PATTERNS.size())
+            .build();
+    }
+
+    @Override
+    public IntegrationPattern create(IntegrationPattern entity) {
```
Comment (jimmidyson) [inline api/src/main/java/com/redhat/ipaas/api/v1/rest/IntegrationPatternDAO.java]: Shame we have to add this all in here, we can probably tidy up interfaces here somehow, but anyway at least we've done this in the entrypoint to the REST API so these should hopefully never be called.

File: api/src/main/java/com/redhat/ipaas/api/v1/rest/CacheManagerProducer.java (hunk 1 above)
Comment (iocanel) [inline api/src/main/java/com/redhat/ipaas/api/v1/rest/CacheManagerProducer.java]: Yes, we can use deltaspike's configuration value. But let's worry about it when we know what to do with the cache.

Comment (jimmidyson) [inline api/src/main/java/com/redhat/ipaas/api/v1/rest/CacheManagerProducer.java]: OK

**Comment by jimmidyson** [comment]
[merge]

**Pull request merged**